ov/labels.go

```diff
@@ -0,0 +1,237 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ov
+
+import (
+	"encoding/json"
+	"github.com/HewlettPackard/oneview-golang/rest"
+	"github.com/HewlettPackard/oneview-golang/utils"
+	"github.com/docker/machine/libmachine/log"
+)
+
+type AllLabels struct {
+	Category    string        `json:"category,omitempty"`
+	Count       int           `json:"count,omitempty"`
+	Created     string        `json:"created,omitempty"`
+	ETAG        utils.Nstring `json:"eTag,omitempty"`
+	Members     []Member      `json:"members,omitempty"`
+	Modified    string        `json:"modified,omitempty"`
+	NextPageUri utils.Nstring `json:"nextPageUri,omitempty"`
+	PrevPageUri utils.Nstring `json:"prevPageUri,omitempty"`
+	Start       int           `json:"start,omitempty"`
+	Total       int           `json:"total,omitempty"`
+	Type        string        `json:"type,omitempty"`
+	Uri         utils.Nstring `json:"uri,omitempty"`
+}
+
+type Member struct {
+	Category string        `json:"category,omitempty"`
+	Created  string        `json:"created,omitempty"`
+	ETAG     string        `json:"eTag,omitempty"`
+	Modified string        `json:"modified,omitempty"`
+	Name     string        `json:"name,omitempty"`
+	Type     string        `json:"type,omitempty"`
+	Uri      utils.Nstring `json:"uri,omitempty"`
+}
+
+type NewLabels struct {
+	LabelNames  []string      `json:"labels,omitempty"`
+	ResourceUri utils.Nstring `json:"resourceUri,omitempty"`
+}
+
+type AssignedLabel struct {
+	Category    string        `json:"category,omitempty"`
+	Created     string        `json:"created,omitempty"`
+	ETAG        utils.Nstring `json:"eTag,omitempty"`
+	Labels      []Label       `json:"labels,omitempty"`
+	Modified    string        `json:"modified,omitempty"`
+	ResourceUri utils.Nstring `json:"resourceUri,omitempty"`
+	Type        string        `json:"type,omitempty"`
+	Uri         utils.Nstring `json:"uri,omitempty"`
+}
+
+type Label struct {
+	Name string        `json:"name,omitempty"`
+	Uri  utils.Nstring `json:"uri,omitempty"`
+}
+
+// GetAllLabels- fetches all labels
+func (c *OVClient) GetAllLabels(sort string, start string, count string, fields string, namePrefix string) (AllLabels, error) {
+	var (
+		uri       = "/rest/labels"
+		q         = make(map[string]interface{})
+		allLabels AllLabels
+	)
+
+	if sort != "" {
+		q["sort"] = sort
+	}
+
+	if fields != "" {
+		q["fields"] = fields
+	}
+
+	if namePrefix != "" {
+		q["namePrefix"] = namePrefix
+	}
+
+	if start != "" {
+		q["start"] = start
+	}
+
+	if count != "" {
+		q["count"] = count
+	}
+
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+
+	// Setup query
+	if len(q) > 0 {
+		c.SetQueryString(q)
```

Comment:
Since you are making changes to ```RestAPICall``` method in one of your as part of issue #315 , why don't you follow the same approach for all the remaining resources..?
```data, err := c.RestAPICall(rest.GET, uri, nil, q)```

Reply:
as discussed will consider this in next epic.